From: Kurt Garloff <garloff@suse.de>
Subject: suid-dumpable ended up in wrong sysctl dir

Diffing in sysctl.c is tricky, using more context is recommended.
suid_dumpable ended up in fs/ instead of kernel/ and the reason
is likely a patch with too little context.

NOTE: This has been in the wrong dir fs/ since it was introduced by
Alan Cox into mainline on 2005-06-23. However, SUSE shipped it
in the correct directory kernel/ in SLES9.

Signed-off-by: Kurt Garloff <garloff@suse.de>

Index: linux-2.6.15/kernel/sysctl.c
===================================================================
--- linux-2.6.15.orig/kernel/sysctl.c
+++ linux-2.6.15/kernel/sysctl.c
@@ -647,12 +647,20 @@ static ctl_table kern_table[] = {
 		.data		= &randomize_va_space,
 		.maxlen		= sizeof(int),
 		.mode		= 0644,
 		.proc_handler	= &proc_dointvec,
 	},
 #endif
+	{
+		.ctl_name	= KERN_SETUID_DUMPABLE,
+		.procname	= "suid_dumpable",
+		.data		= &suid_dumpable,
+		.maxlen		= sizeof(int),
+		.mode		= 0644,
+		.proc_handler	= &proc_dointvec,
+	},
 #if defined(CONFIG_S390) && defined(CONFIG_SMP)
 	{
 		.ctl_name	= KERN_SPIN_RETRY,
 		.procname	= "spin_retry",
 		.data		= &spin_retry,
 		.maxlen		= sizeof (int),
@@ -1032,20 +1040,12 @@ static ctl_table fs_table[] = {
 		.procname	= "inotify",
 		.mode		= 0555,
 		.child		= inotify_table,
 	},
 #endif	
 #endif
-	{
-		.ctl_name	= KERN_SETUID_DUMPABLE,
-		.procname	= "suid_dumpable",
-		.data		= &suid_dumpable,
-		.maxlen		= sizeof(int),
-		.mode		= 0644,
-		.proc_handler	= &proc_dointvec,
-	},
 	{ .ctl_name = 0 }
 };
 
 static ctl_table debug_table[] = {
 	{ .ctl_name = 0 }
 };
